HB/common/database.elpi

```diff
@@ -133,6 +136,21 @@ factories-provide FLwP MLwP :- std.do! [
   w-params.map UnsortedMLwP (p\t\ toposort-mixins) MLwP,
 ].
 
+
+pred undup-terms i:term, i:list term, o:list term.
```

Comment:
This code does not look right.
I guess the simplest is to use `cmp_term` ( https://github.com/LPCIC/coq-elpi/blob/efcf63c433834592beb221c09a5e76f9a847128a/elpi-builtin.elpi#L342 ) in conjunction with the set library ( https://github.com/LPCIC/coq-elpi/blob/efcf63c433834592beb221c09a5e76f9a847128a/elpi-builtin.elpi#L1016 ) and then let the set library do the work. It should work reasonably fine since your terms are ground.